### PR TITLE
Update api key and backend basic auth docs

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md [hunk 1]
```diff
@@ -4,96 +4,93 @@ API Key Authentication in WSO2 Choreo Connect is a simple authentication scheme
...
-## Step 2 - Generate an API Key
-
-Follow the instructions below to generate an API Key via WSO2 API Manager Developer Portal:
-
-{!includes/design/generate-api-key.md!}
-
-## Step 3 - Invoke the API using the API Key
+    When you enable API authentication for an API and deploy via WSO2 API Manager, it will override the API Key Authentication defined in the OpenAPI definition.
+    Therefore, the following steps work for Choreo Connect when the API is not deployed via WSO2 API Manager. 
+    So use the following steps when you want to enable API Key authentication when deploying an API using the APICTL.
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md:17]:
```suggestion
    Therefore, the following steps are only for Choreo Connect standalone mode and would not work when the API is deployed via WSO2 API Manager. Hence, use the following steps when you want to enable API Key authentication when deploying an API using APICTL.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md [hunk 2]
```diff
@@ -4,96 +4,93 @@ API Key Authentication in WSO2 Choreo Connect is a simple authentication scheme
 
 Follow the instructions below to use API Key Authentication with WSO2 Choreo Connect:
 
-## Step 1 - Define API Key Authentication for the API
+- If you are using API Manager to deploy the API, follow [API Key Authentication with WSO2 API Manager]({{base_path}}/design/api-security/api-authentication/secure-apis-using-api-keys).
+- If you are using APICTL to deploy the API, follow [API Key Authentication using OpenAPI security schemes](#api-key-authentication-using-openapi-security-schemes).
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md:9]:
```suggestion
- If you are using APICTL to deploy the API, follow [API Key Authentication for Standalone Mode](#api-key-authentication-for-standalone-mode).

```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md [hunk 3]
```diff
@@ -4,96 +4,93 @@ API Key Authentication in WSO2 Choreo Connect is a simple authentication scheme
...
 
-Use one of the following two methods to enable API Key Authentication in Choreo Connect.
+## API Key Authentication using OpenAPI security schemes
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md:10]:
```suggestion
## API Key Authentication for Standalone Mode
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md [hunk 4]
```diff
@@ -4,96 +4,93 @@ API Key Authentication in WSO2 Choreo Connect is a simple authentication scheme
...
-             security:
-              - ApiKeyAuth: [] 
-             ```
-
-### Method 2 - Using WSO2 API Manager
-
-Follow the instructions below to enable API Key Authentication in WSO2 API Manager via the Publisher Portal.
-
-{!includes/design/create-publish-api.md!}
+Follow the instructions below to enable API Key Authentication using OpenAPI security schemes.
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md:12]:
```suggestion
Follow the instructions below to enable API Key Authentication for standalone mode using OpenAPI security schemes.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md [hunk 5]
```diff
@@ -4,96 +4,93 @@ API Key Authentication in WSO2 Choreo Connect is a simple authentication scheme
...
+    Therefore, the following steps work for Choreo Connect when the API is not deployed via WSO2 API Manager. 
+    So use the following steps when you want to enable API Key authentication when deploying an API using the APICTL.
+
+### Step 1  - Define the API Key security scheme
+
+Security schemes must be defined on the OpenAPI definition under `securitySchemes`. One or more API Key security schemes can be used simultaneously, similar to a logical OR operator. Define the API Key security scheme as follows:
+
+| Field Name | Description   |
+| -----------| --------------|
+| `name`     | Specify a unique name. This name will be used to refer to the scheme on API level or resource level. |
```
Comment (suksw) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md:25]:
```suggestion
| `name`     | Name of the header or query|
```

File: en/docs/includes/design/generate-api-key.md [hunk 6]
```diff
@@ -18,15 +18,15 @@
 
 5. Click **Subscribe**.
 
-     [![Subscribe to the API](https://apim.docs.wso2.com/en/4.1.0/assets/img/learn/subscribe-to-api.png)](https://apim.docs.wso2.com/en/4.1.0/assets/img/learn/subscribe-to-api.png)
+     [![Subscribe to the API]({{base_path}}/assets/img/learn/subscribe-to-api.png)]({{base_path}}/assets/img/learn/subscribe-to-api.png)
```
Comment (samgnaniah) [inline en/docs/includes/design/generate-api-key.md:21]: I don't think base_path works for content like this where you are pulling it into other pages. Please check and revert.

File: en/docs/includes/design/additional-api-key.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+
+#### 1) IP address restriction
+
+The IP address restriction allows only the clients with specific IP addresses can use the token. The IP addresses can be specified
```
Comment (samgnaniah) [inline en/docs/includes/design/additional-api-key.md:4]:
```suggestion
The IP address restriction allows only the clients with specific IP addresses to use the token. The IP addresses can be specified
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/endpoints/defining-a-backend-security-scheme.md [hunk 8]
```diff
@@ -1,25 +1,27 @@
...
-        type: "BASIC"
-        username: "admin"
-        password: "admin"
-  ```
+## Secure Endpoint with Basic Auth
+
+When an actual backend service of the API is protected using basic authentication, the basic authentication parameters (username and password) must be sent to the backend. 
+Hence, you need to define the endpoint security parameter in the OpenAPI using extensions.
+
+The security scheme used for the API must be defined in the `securityConfig` under `x-wso2-production-endpoints` and `x-wso2-sandbox-endpoints vendor extension in the API definition.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/endpoints/defining-a-backend-security-scheme.md:8]:
```suggestion
The security scheme used for the API must be defined in the `securityConfig` under `x-wso2-production-endpoints` and `x-wso2-sandbox-endpoints` vendor extension in the API definition.
```

File: en/docs/includes/design/generate-api-key.md (hunk 6 above)
Comment (AmaliMatharaarachchi) [reply en/docs/includes/design/generate-api-key.md]: Fixed with https://github.com/wso2/docs-apim/pull/5111/commits/29ea6fcfcb2104f29dde05839dc32297d6ac56dc

**Comment by AmaliMatharaarachchi** [comment]
@samgnaniah fixed suggestions

**Pull request merged**